services/filesstore/s3store.go

```diff
@@ -238,10 +238,15 @@ func (b *S3FileBackend) ListDirectory(path string) (*[]string, *model.AppError)
 	}
 
 	doneCh := make(chan struct{})
-
 	defer close(doneCh)
 
-	for object := range s3Clnt.ListObjects(b.bucket, path, false, doneCh) {
+	updatedPath := path
+	if !strings.HasSuffix(path, "/") && len(path) > 0 {
+		// s3Clnt returns only the path itself when "/" is not present
+		// appending "/" to make it consistent across all filesstores
+		updatedPath = path + "/"
```

Comment:
I was going to comment about this being potentially be the wrong path separator for some file systems or OSes but that would only be a consideration in the local file store